libraries/AP_Motors/AP_Motors_Class.cpp

```diff
@@ -198,7 +198,12 @@ void AP_Motors::add_motor_num(int8_t motor_num)
     // ensure valid motor number is provided
     if( motor_num >= 0 && motor_num < AP_MOTORS_MAX_NUM_MOTORS ) {
         uint8_t chan;
-        SRV_Channel::Aux_servo_function_t function = (SRV_Channel::Aux_servo_function_t)(SRV_Channel::k_motor1+motor_num);
+        SRV_Channel::Aux_servo_function_t function;
+        if (motor_num < 8) {
+            function = (SRV_Channel::Aux_servo_function_t)(SRV_Channel::k_motor1+motor_num);
+        } else {
+            function = (SRV_Channel::Aux_servo_function_t)(SRV_Channel::k_motor9+(motor_num-8));
+        }
         SRV_Channels::set_aux_channel_default(function, motor_num);
         if (SRV_Channels::find_channel((SRV_Channel::Aux_servo_function_t)(SRV_Channel::k_motor1+motor_num),
```

Comment:
I think that processing to specify the channel number of the servo assigned to motor 9 - 12 is necessary.